src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -29,10 +29,10 @@ internal TagHelperOutput(string tagName)
         /// <param name="attributes">The HTML attributes.</param>
         public TagHelperOutput(
             string tagName,
-            [NotNull] IDictionary<string, object> attributes)
+            [NotNull] TagHelperAttributes attributes)
         {
             TagName = tagName;
-            Attributes = new Dictionary<string, object>(attributes, StringComparer.OrdinalIgnoreCase);
+            Attributes = attributes;
```

Comment:
we were careful to create a new `Dictionary`.  why is that no longer necessary?  certainly makes debugging more difficult if we can't look at `TagHelperExecutionContext.HTMLAttributes` to see the original state.